src/test/java/com/saralein/server/controller/form/FormPutControllerTest.java

```diff
@@ -0,0 +1,113 @@
+package com.saralein.server.controller.form;
+
+import com.saralein.server.Controller.form.FormPutController;
+import com.saralein.server.data.FormStore;
+import com.saralein.server.request.Request;
+import com.saralein.server.response.Header;
+import com.saralein.server.response.Response;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import static org.junit.Assert.*;
+import org.junit.Before;
+import org.junit.Test;
+
+public class FormPutControllerTest {
+    private FormStore formStore;
+    private FormPutController formPutController;
+
+    @Before
+    public void setUp() {
+        formStore = new FormStore();
+
+        formStore.addData("/form", new LinkedHashMap<String, String>(){{
+            put("My", "Cat");
+            put("More", "Stuff");
+        }});
+
+        formPutController = new FormPutController(formStore);
+    }
+
+    @Test
+    public void updatesExistingDataForSingleData() {
+        Request request = new Request(new HashMap<String, String>(){{
+            put("method", "PUT");
+            put("uri", "/form");
+            put("version", "HTTP/1.1");
+            put("body", "My=Data");
+        }});
+
+        formPutController.createResponse(request);
+        HashMap<String, String> data = formStore.retrieveData("/form");
+
+        assertTrue(data.containsKey("My"));
+        assertEquals("Data", data.get("My"));
+        assertTrue(data.containsKey("More"));
+        assertEquals("Stuff", data.get("More"));
+    }
+
+    @Test
+    public void returnsCorrectResponseForSingleData() {
+        Request request = new Request(new HashMap<String, String>(){{
+            put("method", "PUT");
+            put("uri", "/form");
+            put("version", "HTTP/1.1");
+            put("body", "My=Data");
+        }});
```

Comment:
You have probably already considered this, but in case you haven't, you might think about adding some sort of a request builder. It doesn't feel very "Java-y". Know what I mean?

(It kind of feels like shoehorning dynamics types in 😱 )

Reply:
But not exactly because they are all one type 😄

Reply:
I attempted it once and ended up with something that didn't feel like a builder being used in the parser.  I'm thinking I'll give it another go on a PR after pulling out the application.